libsolidity/interface/CompilerStack.cpp

```diff
@@ -1412,7 +1420,7 @@ void CompilerStack::compileContract(
 
 	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
 
-	std::shared_ptr<Compiler> compiler = std::make_shared<Compiler>(m_evmVersion, m_revertStrings, m_optimiserSettings);
+	std::shared_ptr<Compiler> compiler = std::make_shared<Compiler>(m_evmVersion, m_eofVersion, m_revertStrings, m_optimiserSettings);
```

Comment:
Instead of passing ``m_eofVersion`` around here, we can just assert
```
solUnimplementedAssert(!m_eofVersion, "EOF is not supported for legacy code generation");
```

(this is only-legacy, see the ``solAssert(!m_viaIR, "");`` below)

Reply:
Addressed in https://github.com/ethereum/solidity/pull/15356/files#diff-0f19df92d55a7d7fe38fdf4be1e14b364671c4934853d13a33b31aedf3f917bdR1424